client/cpp/kserve-api/samples/grpc_infer_resnet.cpp

```diff
@@ -0,0 +1,187 @@
+        if (!err.IsOk()) {                                               \
+            std::cerr << "error: " << (MSG) << ": " << err << std::endl; \
+            exit(1);                                                     \
+        }                                                                \
+    }
+
+std::vector<uint8_t> Load(const std::string& fileName) {
+    std::ifstream fileImg(fileName, std::ios::binary);
+    fileImg.seekg(0, std::ios::end);
+    int bufferLength = fileImg.tellg();
+    fileImg.seekg(0, std::ios::beg);
+
+    char* buffer = new char[bufferLength];
+    fileImg.read(buffer, bufferLength);
+
+    return std::vector<uint8_t>(buffer, buffer + bufferLength);
+}
+
+int main(int argc, char** argv) {
+    cxxopts::Options opt("grpc_infer_resnet", "Sends requests via KServe gRPC API.");
+
+    // clang-format off
+    opt.add_options()
+    ("h,help", "Show this help message and exit")
+    ("images_list", "Path to a file with a list of labeled images. ", cxxopts::value<std::string>(), "IMAGES")
+    ("labels_list", "Path to a file with a list of labels. ", cxxopts::value<std::string>(), "LABELS")
+    ("grpc_address", "Specify url to grpc service. ", cxxopts::value<std::string>()->default_value("localhost"), "GRPC_ADDRESS")
+    ("grpc_port", "Specify port to grpc service. ", cxxopts::value<std::string>()->default_value("9000"), "PORT")
+    ("input_name", "Specify input tensor name. ", cxxopts::value<std::string>()->default_value("0"), "INPUT_NAME")
+    ("output_name", "Specify input tensor name. ", cxxopts::value<std::string>()->default_value("1463"), "OUTPUT_NAME")
+    ("model_name", "Define model name, must be same as is in service. ", cxxopts::value<std::string>()->default_value("resnet"), "MODEL_NAME")
+    ("model_version", "Define model version.", cxxopts::value<std::string>(), "MODEL_VERSION")
+    ("timeout", "Request timeout.", cxxopts::value<int>()->default_value("0"), "TIMEOUT")
+    ;
+    // clang-format on
+
+    auto args = opt.parse(argc, argv);
+
+    if (args.count("help")) {
+        std::cout << opt.help() << std::endl;
+        exit(0);
+    }
+
+    std::string input_name(args["input_name"].as<std::string>());
+    std::string output_name(args["output_name"].as<std::string>());
+
+    std::string url(args["grpc_address"].as<std::string>() + ":" + args["grpc_port"].as<std::string>());
+    std::string model_name = args["model_name"].as<std::string>();
+
+    // Create a InferenceServerGrpcClient instance to communicate with the
+    // server using gRPC protocol.
+    std::unique_ptr<tc::InferenceServerGrpcClient> client;
+
+    FAIL_IF_ERR(
+        tc::InferenceServerGrpcClient::Create(&client, url),
+        err);
+
+    std::string img;
+    int label = -1;
+    std::vector<std::string> imgs;
+    std::vector<int> labels;
+    if(args.count("image_list")){
+        std::cout<<"error: option \"images_list\" has no value\n";
+        exit(1);
```

Comment:
Are we switching to return 0? Are you sure we run all destructors when running exit(1) manually?